EventFilter/TotemRawToDigi/interface/SimpleVFATFrameCollection.h

```diff
@@ -0,0 +1,72 @@
+/****************************************************************************
+*
+* This is a part of the TOTEM offline software.
+* Authors: 
+*   Mate Csanad (mate.csanad@cern.ch)
+*   Jan Kašpar (jan.kaspar@gmail.com) 
+*    
+****************************************************************************/
+
+
+#ifndef EventFilter_TotemRawToDigi_SimpleVFATFrameCollection
+#define EventFilter_TotemRawToDigi_SimpleVFATFrameCollection
+
+#include "EventFilter/TotemRawToDigi/interface/VFATFrameCollection.h"
+
+#include <map>
+
+/**
+ * A basic implementation of VFAT frame collection, as map: TotemFramePosition --> VFATFrame.
+**/
+class SimpleVFATFrameCollection : public VFATFrameCollection
+{
+  protected:
+    typedef std::map<TotemFramePosition, VFATFrame> MapType;
+
+    MapType data;
+
+    virtual value_type BeginIterator() const;
+    virtual value_type NextIterator(const value_type&) const;
+    virtual bool IsEndIterator(const value_type&) const;
+
+  public:
+    SimpleVFATFrameCollection();
+    ~SimpleVFATFrameCollection();
+
+    virtual std::string GetClassName() const
+    {
+      return "SimpleVFATFrameCollection";
+    }
+
+    const VFATFrame* GetFrameByID(unsigned int ID) const;
+    const VFATFrame* GetFrameByIndex(TotemFramePosition index) const;
+
+    virtual unsigned int Size() const
+    {
+      return data.size();
+    }
+
+    virtual bool Empty() const
+    {
+      return (data.size() == 0);
+    }
+
+    void Insert(const TotemFramePosition &index, const VFATFrame &frame)
+    {
+      data.insert(std::pair<TotemFramePosition, VFATFrame>(index, frame));
```

Comment:
MapType::value_type (expands to pair) may help to have more generic code
 ... also I think ``` data.insert({index, frame}) ``` should work